lib/galaxy/webapps/galaxy/api/remote_files.py

```diff
@@ -123,15 +126,16 @@ def __load_all_filenames(self, directory):
         response = sorted(response, key=itemgetter("path"))
         return response
 
-    def __create_jstree(self, directory, disable='folders'):
+    def __create_jstree(self, directory, disable='folders', whitelist=None):
         """
         Loads recursively all files and folders within the given folder
         and its subfolders and returns jstree representation
         of its structure.
         """
-        userdir_jstree = None
         jstree_paths = []
-        if os.path.exists(directory) and not os.path.islink(directory):
+        if os.path.exists(directory):
+            if os.path.islink(directory) and not self.__safe_directory(directory, whitelist):
+                raise exceptions.ConfigDoesNotAllowException('directory (%s) is a symlink to a location not on the whitelist' % directory)
             for (dirpath, dirnames, filenames) in os.walk(directory):
```

Comment:
I think the above would not be needed if this used `galaxy.util.path.unsafe_walk()` with the whitelist.